Update reference usage example and fix broken links in configuring a resource doc (#403)
crossplane/upjet: docs/configuring-a-resource.md

}
```

> [!Warning]

> > Please note the `Type` field will be deprecated soon, use `TerraformName` instead.

Reviewer: nit: We may give some motivation for the deprecation.
```suggestion
> Please note the `Reference.Type` field will be deprecated soon, use `Reference.TerraformName` instead. `TerraformName` is a more stable and less error prone API compared to `Type` because it automatically accounts for the configuration changes affecting the cross-resource reference target's kind name, group or version.
```

Author: Done, thank you!

func Configure(p *config.Provider) {
    p.AddResourceConfigurator("aws_iam_access_key", func (r *config.Resource) {
        r.References["user"] = config.Reference{
            TerraformName: "aws_iam_user",
        }
    })
}
```

> Please note the value of `TerraformName` field needs to be a string representing
> the name of the Terraform resource. The utilization of the `TerraformName` field
> remains consistent despite the referenced and referencing resources being in
> different groups. Referencing to a [kms key] from `aws_ebs_volume` resource
> is a good example here:

Reviewer 2: Suggestion: The following reads more natural to me, but please make sure that I'm not glossing over nuances 🙂

```suggestion
`TerraformName` is the name of the Terraform resource, such as
`aws_iam_user`. Because `TerraformName` uniquely identifies a
Terraform resource, it remains the same when the referenced and
referencing resources are in different groups. A good example is
referencing a [kms key] from `aws_ebs_volume` resource:
```

[handle dependencies]: https://docs.crossplane.io/master/concepts/managed-resources/#referencing-other-resources
[user]: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_access_key#user
[generate reference resolution methods]: https://github.com/crossplane/crossplane-tools/pull/35

> [configuration]: https://github.com/crossplane/upjet/blob/942508c5370a697b1cb81d074933ba75d8f1fb4f/pkg/config/resource.go#L172

Reviewer 2: Thanks for linking to specific commit trees, instead of the main branch 🙏 Doing so prevents links going out of context, when the linked resources are updated.
